gcloud-java-examples/src/main/java/com/google/cloud/examples/storage/StorageExample.java

```diff
@@ -512,6 +563,188 @@ public String params() {
     }
   }
 
+  private abstract static class AclAction extends StorageAction<Tuple<BlobId, Acl>> {
+
+    @Override
+    public void run(Storage storage, Tuple<BlobId, Acl> params) {
+      BlobId blobId = params.x();
+      Acl acl = params.y();
+      if (blobId.name().isEmpty()) {
+        Bucket bucket = storage.get(blobId.bucket());
+        if (bucket == null) {
+          System.out.printf("Bucket %s does not exist%n", blobId.bucket());
+          return;
+        }
+        bucket.toBuilder().acl(addAcl(bucket.acl(), acl)).build().update();
+        System.out.printf("Added ACL %s to bucket %s%n", acl, blobId.bucket());
+      } else {
+        Blob blob = storage.get(blobId);
+        if (blob == null) {
+          System.out.printf("Blob %s does not exist%n", blobId);
+          return;
+        }
+        blob.toBuilder().acl(addAcl(blob.acl(), acl)).build().update();
+        System.out.printf("Added ACL %s to blob %s%n", acl, blobId);
+      }
+    }
+
+    private static List<Acl> addAcl(List<Acl> acls, Acl newAcl) {
+      List<Acl> newAcls = new LinkedList<>(acls);
+      newAcls.add(newAcl);
+      return newAcls;
+    }
+  }
+
+  /**
+   * This class demonstrates how to add an ACL to a blob or a bucket for a group of users
+   * (identified by the group's email).
+   *
+   * @see <a href="https://cloud.google.com/storage/docs/access-control/lists#permissions">Access
+   *     Control Lists (ACLs)</a>
+   */
+  private static class AddGroupAclAction extends AclAction {
+
+    @Override
+    Tuple<BlobId, Acl> parse(String... args) {
+      if (args.length >= 3) {
+        BlobId blob;
+        int nextArg;
+        if (args.length == 3) {
+          blob = BlobId.of(args[0], "");
+          nextArg = 1;
+        } else if (args.length == 4) {
+          blob = BlobId.of(args[0], args[1]);
+          nextArg = 2;
+        } else {
+          throw new IllegalArgumentException("Too many arguments.");
+        }
+        String group = args[nextArg++];
+        Acl.Role role = Acl.Role.valueOf(args[nextArg]);
+        return Tuple.of(blob, Acl.of(new Acl.Group(group), role));
```

Comment:
It's returning a Tuple, but does the ACL actually get set in GCS?  It may be a side effect I don't understand, but I don't see the blob actually getting the new ACL.